src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexOptions.cs

```diff
@@ -22,5 +22,6 @@ public enum RegexOptions
 
         ECMAScript              = 0x0100, // "e"
         CultureInvariant        = 0x0200,
+        AnyNewLine              = 0x0400, // "a", Treat "$" as (?=[\r\n]|\z)
```

Comment:
As mentioned in the issue `(?=[\r\n]|\z)` matches "any one of \r, \n, or end of input" whereas what is intended is "any one of \r, \n, \r\n, or end of input". Plus, only end of input if Multiline is not also set (?)

Reply:
I think it needs a more precise definition, which I'm not clear about so hopefully someone can help clarify it in discussion here.